lib/constants/cycles.ts

```diff
@@ -1,58 +1,94 @@
 /**
- * Liturgical Day cycle that can be used as metadata for liturgical days.
+ * Represent the two main cycle, that compose the liturgical year in the Roman rite.
  */
 export const ProperCycles = {
+  /**
+   * The "Proper of Time" (or "Temporal") consists of the movable feasts,
+   * most of them keyed to Easter (which falls on a different Sunday every year),
+   * including Ascension, Pentecost, and so on.
+   */
   ProperOfTime: 'PROPER_OF_TIME',
+  /**
+   * The "Proper of Saints" (or "Sanctorale") consists of the fixed feasts,
+   * celebrated on the very same date each year (no matter what the day of the week),
+   * including Christmas and all the saints' days.
+   */
   ProperOfSaints: 'PROPER_OF_SAINTS',
 } as const;
 
 /**
- * Sundays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link ProperCycles} that contains all the possible proper cycles,
+ * that compose the liturgical year in the Roman rite.
  */
-export const SundaysCycles = {
+export const PROPER_CYCLES = Object.values(ProperCycles);
+
+/**
+ * Represent one of the two main cycles (available from {@link ProperCycles}),
+ * that compose the liturgical year in the Roman rite.
+ */
+export type ProperCycle = typeof PROPER_CYCLES[number];
+
+/**
+ * A three-year cycles for the Sunday mass readings (and some solemnities).
+ * The years are designated A, B, or C. Each yearly cycle begins on the first Sunday of Advent.
+ * Year B follows year A, year C follows year B, then back again to A.
+ */
+export const SundayCycles = {
   YearA: 'YEAR_A',
   YearB: 'YEAR_B',
   YearC: 'YEAR_C',
 } as const;
 
 /**
- * Weekdays cycle that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link SundayCycles} that contains all the possible
+ * three-year Sunday cycles.
+ */
+export const SUNDAY_CYCLES = Object.values(SundayCycles);
+
+/**
+ * Represent one of the three-year Sunday cycles (available from {@link SundayCycles}).
+ */
+export type SundayCycle = typeof SUNDAY_CYCLES[number];
+
+/**
+ * A two-year cycle for the weekday mass readings (also called Cycle I and Cycle II).
+ * Odd-numbered years are the year 1 (or Cycle I); even-numbered ones are the year 2 (or Cycle II).
  */
-export const WeekdaysCycles = {
+export const WeekdayCycles = {
   Year1: 'YEAR_1',
   Year2: 'YEAR_2',
 } as const;
 
 /**
- * Psalter weeks that can be used as metadata for liturgical days.
+ * A dynamically generated array from {@link WeekdayCycles} that contains all the possible
+ * two-year weekday cycles.
  */
-export const PsalterWeeksCycles = {
-  Week1: 'WEEK_1',
-  Week2: 'WEEK_2',
-  Week3: 'WEEK_3',
-  Week4: 'WEEK_4',
-} as const;
+export const WEEKDAY_CYCLES = Object.values(WeekdayCycles);
 
 /**
- * A dynamically generated constant consisting of all the enum keys in [[LITURGICAL_DAY_CYCLES]]
+ * Represent one of the two-year weekday cycles (available from {@link WeekdayCycles}).
```

Comment:
```suggestion
 * Represent one of the weekday cycles (available from {@link WeekdayCycles}).
```

Reply:
I still think this suggestion should be applied.

Reply:
I still think that keeping the "two-year" provide a more precise information.

```suggestion
 * Represent one of the two-year weekday cycle (available from {@link WeekdayCycles}).
```

Reply:
LGTM.